### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by arikmaor**
Includes:
1. `react-router`, `material-ui` and `storybook` installation
2. Login page naive implemntation (against real `node-server`)

### CODE REVIEW

File: src/auth/LoginPage.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+export default function LoginPage() {
+  return (
+    <div>
+      Login Page
+    </div>
+  );
+}
```
Comment (AsafYehezkelTL) [inline src/auth/LoginPage.tsx]: There is also the arrow function syntax:
```typescript
export const LoginPage = () => {
   return (
      <div>
        Login Page
      </div>
  );
}
```

Tiny differences that when using functions the caller become the `this` in oppose to arrow function 
WDYT?

File: src/auth/AuthContext.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  login: () => Promise.reject(),
+  logout: () => Promise.reject(),
+})
+
+export const AuthProvider: FC<PropsWithChildren<{}>> = ({children}) => {
+  const [user, setUser] = useState<UserData>()
+
+  useEffect(() => {
+    api.whoAmI().then((loggedInUser) => setUser(loggedInUser)).catch(() => {});
+  });
```
Comment (AsafYehezkelTL) [inline src/auth/AuthContext.tsx:21]: `useEffect` will run async here with no blocking useDemo from being execute.
Do we want it like that? of shall we use `await` here?

File: src/core/api-client.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import {DefaultApi, Configuration} from '@tensorleap/api-client';
+
+const client = new DefaultApi(
+  new Configuration({
+    basePath: `${document.location.origin.replace(/-webapp2/, '-api')}/api/v2`,
+    middleware: [
+      {
+        post: async ({response}) => {
+          if (response.status === 401) {
+            // authModule.logout();
```
Comment (AsafYehezkelTL) [inline src/core/api-client.tsx:10]: why in comment?

File: src/auth/LoginPage.tsx [hunk 4]
```diff
@@ -1,9 +1,31 @@
...
 export default function LoginPage() {
+  const auth = useAuth();
+  const emailRef = useRef<HTMLInputElement>();
+  const passwordRef = useRef<HTMLInputElement>();
+
+  const onLogin = useCallback(async () => {
+    const email = emailRef?.current?.value;
+    const password = passwordRef?.current?.value;
+    if (!email || !password) {
+      return
```
Comment (AsafYehezkelTL) [inline src/auth/LoginPage.tsx:14]: Will it be validated eventually? the required fields prior click login?

File: src/stories/LoginPage.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+export default {
+  title: 'Pages/LoginPage',
+  component: LoginPage,
+  argTypes: {
+    onLogin: {action: 'Login'},
+    onLogout: {action: 'Logout'}
+  }
+} as Meta;
+
+function createMockAuthService(args: any): AuthContextInterface {
```
Comment (AsafYehezkelTL) [inline src/stories/LoginPage.stories.tsx:17]: consider using here object destruction like:
```typescript
{onLogin, onLogout}
```
And adding types rather `any`

File: src/stories/LoginPage.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  return {
+    user: undefined,
+    login: args.onLogin,
+    logout: args.onLogout,
+  }
+}
+
+const Template: Story<{}> = (args) => (
+  <AuthContext.Provider value={createMockAuthService(args)}>
```
Comment (AsafYehezkelTL) [inline src/stories/LoginPage.stories.tsx:26]:
```typescript
<AuthContext.Provider value={() => createMockAuthService(args)}>
```

**Review by AsafYehezkelTL — CHANGES_REQUESTED** [review]

File: src/auth/LoginPage.tsx (hunk 1 above)
Comment (arikmaor) [reply src/auth/LoginPage.tsx]: AFAIK, there is **NO** difference at all between the two in the top scope
Matter of preference

Comment (AsafYehezkelTL) [reply src/auth/LoginPage.tsx]: The differences is the caller but it doesn't critical at all

File: src/auth/AuthContext.tsx (hunk 2 above)
Comment (arikmaor) [reply src/auth/AuthContext.tsx:21]: It's a bit problematic using an `async` function here as `useEffect` expects a function that returns `undefined` or a cleanup function

Said that, we should definitely find the hack that will allow us to refactor and use `async/await` here (Not in this PR unless you think it's a must)

File: src/auth/LoginPage.tsx (hunk 4 above)
Comment (arikmaor) [reply src/auth/LoginPage.tsx:14]: Currently this component is kind of a placeholder, when we have a proper design we can put the validation accordingly

right now I'm just checking that there is an email/password so there are no compilation errors

File: src/core/api-client.tsx (hunk 3 above)
Comment (arikmaor) [reply src/core/api-client.tsx:10]: This is where the old app logged out
Still haven't implemented the auto log out here so I just kept this commented to know this is where I should put the logic

File: src/stories/LoginPage.stories.tsx (hunk 5 above)
Comment (arikmaor) [reply src/stories/LoginPage.stories.tsx:17]: 10x, fixed

File: src/stories/LoginPage.stories.tsx (hunk 6 above)
Comment (arikmaor) [reply src/stories/LoginPage.stories.tsx:26]: nope

File: src/auth/AuthContext.tsx (hunk 2 above)
Comment (AsafYehezkelTL) [reply src/auth/AuthContext.tsx:21]: OK, so just be aware that line #23 will executed before user logged in

File: src/auth/LoginPage.tsx (hunk 4 above)
Comment (AsafYehezkelTL) [reply src/auth/LoginPage.tsx:14]: Cool!

File: src/core/api-client.tsx (hunk 3 above)
Comment (AsafYehezkelTL) [reply src/core/api-client.tsx:10]: Thanks

**Review by AsafYehezkelTL — APPROVED** [review]
Looks Great!

File: src/auth/AuthContext.tsx (hunk 2 above)
Comment (arikmaor) [reply src/auth/AuthContext.tsx:21]: of course...
Then again once the user is set

And if we want to cache the user in the browser, we'll use the cached value in `const [user, setUser] = useState<UserData>(CACHED_VALUE)`
Then user will be from the cache and then removed...

lot's of tricks there

**Pull request merged**